features/filesystem/storagelite/source/StorageLiteFS.cpp

```diff
@@ -0,0 +1,320 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "StorageLiteFS.h"
+#include "mbed_retarget.h"
+#include "string.h"
+#include "stdlib.h"
+
+typedef struct {
+    char *file_name;
+    int open_flags;
+} file_handle_data_t;
+
+// Convert basic StorageLite error numbers to errno based ones
+static int convert_to_errno(int err)
+{
+    switch (err) {
+        case STORAGELITE_SUCCESS:
+            return  0;
+        case STORAGELITE_READ_ERROR:
+            return -EIO;
+        case STORAGELITE_WRITE_ERROR:
+            return -EIO;
+        case STORAGELITE_NOT_FOUND:
+            return -ENOENT;
+        case STORAGELITE_DATA_CORRUPT:
+            return -EILSEQ;
+        case STORAGELITE_BAD_VALUE:
+            return -EINVAL;
+        case STORAGELITE_BUFF_TOO_SMALL:
+            return -ENOMEM;
+        case STORAGELITE_NO_SPACE_ON_BD:
+            return -ENOSPC;
+        case STORAGELITE_OS_ERROR:
+            return -EINTR; // FIXME: Find a better error
+        case STORAGELITE_UNINITIALIZED:
+            return -ENXIO; // FIXME: Find a better error
+        case STORAGELITE_MAX_FILES_REACHED:
+            return -ENFILE;
+        case STORAGELITE_NOT_SUPPORTED:
+            return -ENOSYS;
```

Comment:
Just to add - our FS test initially included an errno check that failed when compiled by the ARM compiler, as the errno numbers didn't match the ones in mbed-os. I believe any test that would try checking errno values would fail the same way.